lib/internal/http2/compat.js

```diff
@@ -550,13 +550,14 @@ class Http2ServerResponse extends Stream {
     if (headers === undefined && typeof statusMessage === 'object')
       headers = statusMessage;
 
-    if (typeof headers === 'object') {
-      const keys = Object.keys(headers);
-      let key = '';
-      for (var i = 0; i < keys.length; i++) {
-        key = keys[i];
-        this[kSetHeader](key, headers[key]);
+    if (typeof headers === 'object' && headers !== null) {
+      for (const key of Object.keys(headers)) {
```

Comment:
This is significantly worse for performance. I'm unclear why this change was made?

Reply:
@apapirovski: It is? As far as I know nowadays there is no diff in V8.

> I'm unclear why this change was made?

Not necessary. Mostly iterative cleanup?